contracts/ArchetypeLogic.sol

```diff
@@ -208,6 +210,12 @@ library ArchetypeLogic {
       revert WalletUnauthorizedToMint();
     }
 
+    if (config.blacklisted != bytes32(0)) {
+      if (isBlacklisted(auth.proof, config.blacklisted, msgSender)) {
```

Comment:
Looking at this some more I see some limitations:

There would be no way to implement both a whitelist and a blacklist in the same contract, as we are using the same proof parameter for both. I can see a common use case as:
  ```
  public black list
  private friends free mint whitelist
  ```
In this scenario the whitelist mint would always fail.

Reply:
Another limitation is we are relying on the user sending an honest proof. The minter can send an invalid proof causing blackList merkle check to fail. Letting them bypass the check.